File: backend/spring-nfc/src/main/java/com/example/springnfc/entity/Intervention.java
package com.example.springnfc.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Collection;
import java.util.Date;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@ToString
public class Intervention implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Date createdAt = new Date();
    private Date updatedAt;
    @Lob
    private String detail;

    @ManyToOne
    private Utilisateur utilisateur;

    @ManyToOne
    private Installation installation;

    @OneToMany(mappedBy="intervention")
    private Collection<Etape> etapes;
}

File: frontend/flutter-nfc/lib/models/intervention.dart
class Intervention {
  int? id;
  String? detail;
  String? name;
  bool? completed;


  Intervention({required this.id, required this.detail, required this.name, required this.completed});

  Intervention.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    detail = json['detail'];
    name = json['name'];
    completed = json['completed'];
  }


  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id.toString();
    data['detail'] = this.detail.toString();
    data['name'] = this.name.toString();
    data['completed'] = this.completed.toString();
    return data;
  }
}